### PR TITLE
feature: call _setMetadata in createDao

### PR DESCRIPTION
**Pull request by aodhgan**
### Purpose for this PR
as per https://github.com/windranger-io/windranger-treasury/issues/359

### CODE REVIEW

File: docs/deploy_local.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ export BOND_MEDIATOR_CONTRACT=${BondMediator}
 export BOND_FACTORY_CONTRACT=${BondFactory}
 export COLLATERAL_TOKENS_CONTRACT=${BitToken}
 export TREASURY_ADDRESS=${Treasury}
+export DAO_METADATA=${metadataString}
```
Comment (CjHare) [inline docs/deploy_local.md:54]: issue: Variable consistency, use PascalCase for `MetadataString`.

_You've got caemCase and the others are all PascalCase

Comment (CjHare) [inline docs/deploy_local.md:54]: issue: Name variable `DaoMetadata` as it is only for the Dao and `String` is both inconsistent and redundant.

_Description for these variables are done further up in the readme_

Comment (CjHare) [inline docs/deploy_local.md:54]: issue: Entry needed in `#### Environment Variables needed for Verify`

You're adding an env var but not adding the description for it.

File: test/contracts/staking/staking-mediator.test.ts [hunk 2]
```diff
@@ -130,7 +131,7 @@ describe('Staking Pool Mediator contract', () => {
                 const expectedDaoId = previousHighestDaoId.add(1).toBigInt()
```
Comment (CjHare) [inline test/contracts/staking/staking-mediator.test.ts:131]: issue: You need t least one new test case: when `METADATA` is  `undefined`.

_Depends on what the expected behaviour is for metadata, whether it's mandatory (where you'll need more) or optional_

File: contracts/staking/StakingPoolMediator.sol [hunk 3]
```diff
@@ -56,14 +56,16 @@ contract StakingPoolMediator is
...
         external
         atLeastDaoCreatorRole
         returns (uint256)
     {
         uint256 id = _daoConfiguration(erc20CapableTreasury);
         _grantDaoCreatorAdminRoleInTheirDao(id);
 
+        _setDaoMetaData(id, metadata);
+
         emit CreateDao(id, erc20CapableTreasury, _msgSender());
```
Comment (CjHare) [inline contracts/staking/StakingPoolMediator.sol:69]: question: should there be the metadata in the emit?

Comment (aodhgan) [reply contracts/staking/StakingPoolMediator.sol:69]: https://github.com/windranger-io/windranger-treasury/blob/main/contracts/dao-configuration/DaoConfiguration.sol#L91 emits it

File: test/contracts/staking/staking-mediator.test.ts (hunk 2 above)
Comment (aodhgan) [reply test/contracts/staking/staking-mediator.test.ts:131]: Ok i've added one - `set as empty string`. Is this what you mean or you are talking about calling `createDao()` with an empty string?

Comment (CjHare) [reply test/contracts/staking/staking-mediator.test.ts:131]: Yes, invoking `createDao()` with an empty string.

**Comment by aodhgan** [comment]
replaced by https://github.com/windranger-io/windranger-treasury/pull/376

**Pull request closed without merging**